client/components/sub-masterbar-nav/index.jsx

```diff
@@ -0,0 +1,161 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import { findDOMNode } from 'react-dom';
+import classNames from 'classnames';
+
+import Gridicon from 'gridicons';
+import Item from './item';
+
+const VIEW_BREAKPOINT = 660;
+const VIEW_PADDING = 50;
+const ITEM_WIDTH = 110;
+
+export default class SubMasterbarNav extends Component {
+	static propTypes = {
+		fallback: PropTypes.shape(
+			{
+				label: PropTypes.string.isRequired,
+				uri: PropTypes.string.isRequired,
+				icon: PropTypes.string
+			}
+		),
+		options: PropTypes.arrayOf(
+			PropTypes.shape( {
+				label: PropTypes.string.isRequired,
+				uri: PropTypes.string.isRequired,
+				icon: PropTypes.string
+			} )
+		),
+		uri: PropTypes.string.isRequired
+	};
+
+	static defaultProps = {
+		options: []
+	}
+
+	state = {
+		collapsed: true
+	};
+
+	componentDidMount() {
+		this.onResize();
+
+		window.addEventListener( 'resize', this.onResize.bind( this ) );
```

Comment:
That makes sense. Should I do this just for the functions that need it or all of them for better readability ?

Also, how should I solve the issue with ```onClick={ this.selectItem.bind( this, item ) }``` ?  
I have a feeling I should either be passing the whole object to the ```Item``` component and have it pass it back as an argument, or I should be passing indexes instead of options into ```selectItem```.